[prometheus] Add individual nodeSelector and toleration settings for longterm (#3711)
deckhouse/deckhouse: docs/documentation/pages/internal/DEVELOPMENT.md

**If the bug is not urgent, you better be patient and wait until the fix is implemented as part of the standard release process. This is especially true for the `Stable` and `Rock Solid` release channels.**

Periodicity and timing of patch-release updates in the release channels:
1. The Deckhouse team can change patch-release versions in the `Alpha` channel at any time at its own discretion with any periodicity and without prior warning.
2. For the `Beta` and `Early Access` channels, the Deckhouse team can switch to a patch-release version at any time at its own discretion but not earlier than two hours after switching to this version in the `Alpha` and `Beta` release channels, accordingly.
3. For the `Stable` and `Rock Solid` channels, the switching to the patch-release version can be performed on any day:
   1. `Stable`, `Rock Solid` — not earlier than the next day after the switching to this version in the `Early Access` and `Stable` release channels, accordingly.

> 4. These rules may be breached if there is a **real need** to make urgent changes due to a critical bug/vulnerability (note that all actions **must be coordinated with the team leaders**).

Reviewer: ```suggestion
4. These rules may be breached if there is a **real need** to make urgent changes due to a critical bug/vulnerability (do these actions **at your own risk**).
```

Author: Sorry, this is not related to the current changes anymore. Not applicable to this PR.

2. `monitoring` - uses the same logic as the `system` and `frontend` strategies but includes an extra step after all of the above:
    * If nodes with the `node-role.deckhouse.io/system=""` label are found in the cluster, then this value is used as the nodeSelector. It is assumed that if there are no dedicated monitoring nodes in the cluster, then the components of monitoring-related modules run on the system nodes.

3. `monitoring-longterm` - uses the same logic as the `system` and `frontend` strategies but includes an extra step before all of the above:

>     * If a longtermNodeSelector is defined, it will be used, then node with the name of the strategy will be validated, otherwise continue with logic described above in the `monitoring` section.

Reviewer 2: ```suggestion
    * If a `longtermNodeSelector` is defined, it will be used, then node with the name of the strategy will be validated, otherwise continue with logic described above in the `monitoring` section.
```

    value: "system"
  ```

* For the `monitoring-longterm` strategy, the rules will look as follows:

  ```yaml
  tolerations:
  - key: dedicated.deckhouse.io
    operator: Equal

>     value: {{ .Chart.Name }}-longterm

Reviewer 2: It is weird to see `{{ .Chart.Name }}` here, ptobably, there was an error before.

Author: There was the same as for monitoring strategy as 
  ```yaml
  tolerations:
  - key: dedicated.deckhouse.io
    operator: Equal
    value: {{ .Chart.Name }}
  - key: dedicated.deckhouse.io
    operator: Equal
    value: {{ strategy_name }}
  - key: dedicated.deckhouse.io
    operator: Equal
    value: "system"
  ```

is not it?

Author: Section removed with approach. Please review

#### kubectl

> We do not recommend using kubectl in hooks. It leads to a loss of idempotency since the hook depends on the cluster state in addition to the input parameters (that creates some difficulties during debugging/testing).

Reviewer 2: create an extra pr for this fix please.

Author: Got it, fixed.
